src/Adapter/Product/Image/Repository/ProductImageMultiShopRepository.php

```diff
@@ -285,4 +291,105 @@ function (int $shopId, array $productImages): ShopProductImages {
 
         return ShopProductImagesCollection::from(...$shopProductImagesArray);
     }
+
+    public function getCoverImageId(ProductId $productId, ShopId $shopId): ?ImageId
+    {
+        $result = $this->connection->createQueryBuilder()
+            ->select('id_image')
+            ->from($this->dbPrefix . 'image_shop')
+            ->where('id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+            ->andWhere('id_shop = :shopId')
+            ->setParameter('shopId', $shopId->getValue())
+            ->andWhere('cover = 1')
+            ->execute()
+            ->fetchOne()
+            ;
+
+        return $result ? new ImageId((int) $result) : null;
+    }
+
+    public function associateImageToShop(Image $image, ShopId $shopId): void
+    {
+        $this->connection->createQueryBuilder()
+            ->insert($this->dbPrefix . 'image_shop')
+            ->values(
+                [
+                    'id_product' => ':productId',
+                    'id_image' => ':imageId',
+                    'id_shop' => ':shopId',
+                    'cover' => ':cover',
+                ]
+            )
+
+            ->setParameter('productId', (int) $image->id_product)
+            ->setParameter('imageId', (int) $image->id)
+            ->setParameter('shopId', $shopId->getValue())
+            ->setParameter('cover', $image->cover ? 1 : null)
+            ->execute()
+        ;
+    }
+
+    public function setCoversIfDoesntExist(ProductId $productId): void
+    {
+        //todo: remove duplication
+        $results = $this->connection->createQueryBuilder()
+            ->select('id_image', 'id_shop', 'cover')
+            ->from($this->dbPrefix . 'image_shop', 'i')
+            ->andWhere('i.id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+            ->addOrderBy('i.id_image', 'ASC')//todo: to remove
```

Comment:
I believe in the behat tests you mention the cover should be the first in position I agree it makes more sense But then I believe you should order by `i.position` here no?